### PR TITLE
Getting Started With NLP Transformers

### LINKED ISSUE #5338: [Machine Learning] Getting Started with NLP Transformers
### Topic Suggestion
Getting Started with NLP Transformers

### Writing sample(s): 
N/A

#### Proposed article introduction
A transformer is a deep learning model that adopts the mechanism of self-attention, differentially weighting the significance of each part of the input data. It is mostly used in the field of natural language processing (NLP) and in computer vision (CV). Transformers are mainly used in areas like machine language translation, conversational chatbots, and powering better search engines. 

Natural language processing()NLP) is a subfield of linguistics and artificial intelligence concerned with the interactions between computers and human language, i.e programming computers to process and analyze large amounts of natural language data.

In this tutorial, we will look at transformers in NLP and their advantages over the traditional deep learning algorithms like recurrent neural networks (RNN). We will then build an NLP model using a transformer.

#### Key takeaways 

- Understanding the Model Architecture of the Transformer in NLP
- Understanding the working principles behind Transformers
- Modeling NLP using a transformer.

#### Article quality
This article gives in-depth information and handy tips on what has already been covered.

#### References
N/A

### PR DESCRIPTION
**Pull request by norah254**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #5338 

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

File: content/articles/getting-started-with-nlp-transformers/index.md [hunk 1]
```diff
@@ -0,0 +1,193 @@
+﻿
+### Getting Started with NLP Transformers
+In today's world of daily new improvements and developments in technology, the field of Natural Language Processing (NLP) has not been left behind. New developments like Transfer Learning, Transformers (e.g., Google BERT & ELMO), Reinforcement Learning, Low-Code Tools etc. have changed the way data scientist work with textual data. They have also helped increase the area application for NLP like detecting fake news and monitoring social media for cases of cyber bullying.
```
Comment (jumamiller) [inline content/articles/getting-started-with-nlp-transformers/index.md:3]: Kindly break this down into 2 paragraphs.

File: content/articles/getting-started-with-nlp-transformers/index.md [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+
+In this article, we will look at Transformers in NLP and their advantages over the traditional deep learning algorithms like recurrent neural networks (RNN). We will then learn how to build an NLP model using a transformer.
+
+### Introduction
+
```
Comment (jumamiller) [inline content/articles/getting-started-with-nlp-transformers/index.md:8]: Ensure this spacing between the heading and first paragraph is removed throughout your article!

**Review by jumamiller — CHANGES_REQUESTED** [review]
Hi, work on those minor edits before we move to tech review!

**Comment by jumamiller** [comment]
The changes have not been reflected on the above commit?

**Comment by norah254** [comment]
Check well, the changes have just not been highlighted.

File: content/articles/getting-started-with-nlp-transformers/index.md [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+### Conclusion
+
+Now that we have looked at the basic working principles of Transformers and built two NLP models using a Transformer and seen the benefits of using Transformers over sequence-to-sequence models like RNN, you can now apply the knowledge into your personal or work projects.
+
+Happy coding!
+
+### More Reading
+
+* Attention is All you need [research paper](https://arxiv.org/pdf/1706.03762.pdf).
+* Model more [NLP Transformer tasks](https://www.kaggle.com/varunyadav17/nlp-transformer-tasks#Sequence-Classification-Task--).
```
Comment (jumamiller) [inline content/articles/getting-started-with-nlp-transformers/index.md:193]: And these?

File: content/articles/getting-started-with-nlp-transformers/index.md [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+result
+```
+The code above results to a negative sentiment based on the word "hate" as seen below:
+
+![negative](/engineering-education/getting-started-with-nlp/negative.jpg)
+
+The next task will be a sequence paraphrasing task.
+
+#### b) Sequence Paraphrasing Task
+
```
Comment (jumamiller) [inline content/articles/getting-started-with-nlp-transformers/index.md:120]: This one too

File: content/articles/getting-started-with-nlp-transformers/index.md (hunk 2 above)
Comment (jumamiller) [reply content/articles/getting-started-with-nlp-transformers/index.md:8]: > Ensure this spacing between the heading and first paragraph is removed throughout your article!

This comment here; FIX THE ENTIRE ARTICLE!

**Comment by jumamiller** [comment]
Hi, are you having challenges making edits? The commits you're making are not resolving the requested changes!

Also, ensure you pass the article on Grammarly, 'cause it has a couple of mistakes.

Kindly reach me on Slack(@Miller) in case you're stuck.

**Comment by WanjaMIKE** [comment]
Plagiarism check is done.

**Review by jumamiller — APPROVED** [review]
Hi @norah254, kindly check if the author details have been correctly entered!

A great article to get you started @section. Kudos!

**Review by louisefindlay23 — CHANGES_REQUESTED** [review]
Please rename the images where you have the file extensions to be block capitals to lowercase.

**Comment by norah254** [comment]
@louisefindlay23 and @jumamiller since GitHub web interface doesn't allow editing of images, Could I close the PR and ping you on a new one once the changes are done?

**Comment by jumamiller** [comment]
@louisefindlay23 I think images should just work fine. I have tested them on my end(Not sure of your opinion on this).

@norah254 to edit, delete the specific image(s) and then re-upload. Do not close the PR.

**Comment by norah254** [comment]
@louisefindlay23 the changes are done

**Comment by jumamiller** [comment]
@louisefindlay23 images have been renamed!

**Comment by jumamiller** [comment]
@ahmadmardeni1, @Kayere  are you in a position to publish this article?

**Comment by WanjaMIKE** [comment]
Merged #5760 into master

**Pull request merged**